bin/ember-template-lint.js

```diff
@@ -245,16 +249,34 @@ async function run() {
     process.exitCode = 1;
   }
 
+  printResults(results, { options, todoInfo });
+}
+
+function printResults(results, { options, todoInfo }) {
   let hasErrors = results.errorCount > 0;
   let hasWarnings = results.warningCount > 0;
   let hasTodos = options.includeTodo && results.todoCount;
   let hasUpdatedTodos = options.updateTodo;
 
-  let printer = new Printer({
+  let formatter = loadFormatter({
     ...options,
     hasResultData: hasErrors || hasWarnings || hasTodos || hasUpdatedTodos,
   });
-  printer.print(results, todoInfo);
+
+  let output =
+    typeof formatter.format === 'function'
+      ? formatter.format(results, todoInfo)
+      : // support legacy formatters
+        formatter.print(results, todoInfo);
+
+  if (output) {
+    if ('output-file' in options) {
+      let outputPath = writeOutputFile(output, formatter.outputFileType || 'txt', options);
+      console.log(`Report written to ${outputPath}`);
+    } else {
+      console.log(output);
+    }
+  }
```

Comment:
Let's tweak this so that we don't accidentally "leak" the ability to return a value to `formatter.print`. I'm thinking we actually want something like (but probably don't take inline suggestion, due to prettier issues):

```suggestion
  if (typeof formatter.format === 'function') {
    if ('output-file' in options) {
      let outputPath = writeOutputFile(output, formatter.outputFileType || 'txt', options);
      console.log(`Report written to ${outputPath}`);
    } else {
      console.log(output);
    }
  } else {
    // support legacy formatters
    formatter.print(results, todoInfo);
  }
```